krabs/krabs/ut.hpp

```diff
@@ -99,42 +105,75 @@ namespace krabs { namespace details {
+                if (filter.provider_filter_event_id() > 0) {
+                	//native id existing, set native filters
+					auto& provider_filter_event_ids = provider_flags[provider.get().guid_].provider_filter_event_ids_;
+					provider_filter_event_ids.push_back(filter.provider_filter_event_id()); 
+                }
+            }
         }
 
-        for (auto &provider : providerFlags) {
-            GUID guid = provider.first;
-
+        for (auto &provider : provider_flags) {
             ENABLE_TRACE_PARAMETERS parameters;
+            parameters.ControlFlags = 0;
             parameters.Version = ENABLE_TRACE_PARAMETERS_VERSION_2;
             parameters.SourceId = provider.first;
-            parameters.EnableProperty = std::get<3>(provider.second);
-            parameters.ControlFlags = 0;
+            
+            GUID guid = provider.first;
+            parameters.EnableProperty = std::get<3>(provider.second.flags_tuple_);
             parameters.EnableFilterDesc = nullptr;
             parameters.FilterDescCount = 0;
+            EVENT_FILTER_DESCRIPTOR filterDesc;
+            std::unique_ptr<BYTE[]> filterMemoryPtr;
+
+            if (provider.second.provider_filter_event_ids_.size() > 0) {
+                //event filters existing, set native filters using API
+                parameters.FilterDescCount = 1;  
+
+                ZeroMemory(&filterDesc, sizeof(filterDesc));
+                filterDesc.Type = EVENT_FILTER_TYPE_EVENT_ID;
+
+                //allocate + size of expected events in filter
+                DWORD size = FIELD_OFFSET(EVENT_FILTER_EVENT_ID, Events[provider.second.provider_filter_event_ids_.size()]);
+                filterMemoryPtr = std::make_unique<BYTE[]>(size);
+
+                auto filterEventIds = reinterpret_cast<PEVENT_FILTER_EVENT_ID>(filterMemoryPtr.get());
+                filterEventIds->FilterIn = TRUE;
+                filterEventIds->Count = static_cast<USHORT>(provider.second.provider_filter_event_ids_.size());
+                for (int index=0;index<filterEventIds->Count;++index) {
+                	filterEventIds->Events[index] = provider.second.provider_filter_event_ids_[index];
+                }
+                filterDesc.Ptr = reinterpret_cast<ULONGLONG>(filterEventIds);
+                filterDesc.Size = size;
+
+                parameters.EnableFilterDesc = &filterDesc;
+            }
 
             ULONG status = EnableTraceEx2(trace.registrationHandle_,
-                                          &guid,
-                                          EVENT_CONTROL_CODE_ENABLE_PROVIDER,
-                                          std::get<0>(provider.second),
-                                          std::get<1>(provider.second),
-                                          std::get<2>(provider.second),
-                                          0,
-                                          &parameters);
+                                        &guid,
```

Comment:
Please maintain the formatting here.

Reply:
Hi, Zac
My apologies, I did not understand what to fix here.
I just compared mentioned by you lines code that existed before I created the branch against mine and I see the same indentation and same style.
Please clarify 
Thanks
Issahar

Reply:
not fixed: please reopen the issue up on the response